game/game/movealgo.cpp

```diff
@@ -44,10 +44,20 @@ void MoveAlgo::tickMobsi(uint64_t dt) {
     return;
 
   auto dp  = animMoveSpeed(dt);
-  if(!npc.interactive()->isLadder())
+  auto pos = npc.position();
+  if(npc.interactive()->isLadder()) {
+    auto rotY=-npc.rotationY();
+    if (std::abs(rotY)>0.001 && std::abs(dp.y)>0.001) {
```

Comment:
What this code do? I've tested without it - no notifiable difference.

Reply:
For tilted ladders. Character fails to properly climb up at top. And if the ladder is attached to a wall, the character climbs into the wall if climbing down.

Reply:
Let's not do it for now - in test Jarkendar ladder works most of the time. Top-most ladder sometimes works poorly, but it same as in vanilla.

Reply:
They shouldn't be affected by it. There are only a few in G1.

Reply:
This is quite awkward code still :(
Why `std::abs`? There is seemingly no divisions downstream or something. And `cos/sin` should be way faster than the `if`

AFAIR this is G1 only, right? Can we decouple this change into another PR. This `*=-2` fell like fine tuning; `Vec3(s,0,-c)` - in  other places it's usually `Vec3(c,0,s)`

Reply:
I only want to test if both values are nonzero, is there a better way? If `rotY=0` only a zero vector is added, so this check could be left out. `std::abs(dp.y)>0.001` is needed to know if the character is moving and don't add anything if not.

`Vec3(s,0,-c)` stuff is copied from `marvinF8/K` routines. Factor `2` is needed to make downstairs climbing look nice and yeah tilted ladders occur only in G1.